app/assets/javascripts/3dbio_viewer/src/webapp/components/molecular-structure/MolecularStructure.tsx

```diff
@@ -87,6 +87,9 @@ function usePdbePlugin(options: MolecularStructureProps) {
     debugVariable({ pdbePlugin });
     const chains = options.pdbInfo?.chains;
 
+    const uploadDataToken = newSelection.type === "uploadData" ? newSelection.token : undefined;
+    const extension = newSelection.type === "uploadData" ? newSelection.extension : "";
```

Comment:
[minor, no need to change] when assignation/conditions go in group, we can write: `const [a, b] = condition ? [1,2] : [3,4]` (we can also use objects instead of tuples)